reference/src/layout/arrays.md

```diff
@@ -0,0 +1,33 @@
+# Layout of Rust array types
+
+Array types, `[T; N]`, store `N` values of type `T` contiguously with a constant
+_stride_ (the distance between each two consecutive values).
+
+The offset of the first array element is `0`.
+
+The stride of the array is computed as the size of the element type rounded up
+to the next multiple of the alignment of the element type. That is, the _stride_
+of an array can be larger than the element size iff the alignment requirements
+of the element are larger than its size.
```

Comment:
I think it is worth mentioning that in current Rust, we always have `size >= align` and hence we always have `stride == size` -- but we might want to keep the door open to change that in the future.

Reply:
Please check if the last commit addresses this.